### PR TITLE
Full stack test

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: t/full-stack.t [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+my $schedulerpid = fork();
+if ($schedulerpid == 0) {
+    use OpenQA::Scheduler;
+    OpenQA::Scheduler::run;
+    Devel::Cover::report() if Devel::Cover->can('report');
+    _exit(0);
+}
+
+# we don't want no fixtures
+sub noop_hook {
```
Comment (Martchus) [inline t/full-stack.t]: Is this actually used?

**Comment by Martchus** [comment]
Getting a lot of warnings when running this locally

```
Use of uninitialized value in concatenation (.) or string at template
        test/scheduled_table.html.ep line 44 (#1)
Use of uninitialized value $build in concatenation (.) or string at template
        test/scheduled_table.html.ep line 46 (#1)
Use of uninitialized value $distri in concatenation (.) or string at template
        test/scheduled_table.html.ep line 47 (#1)
Use of uninitialized value $flavor in concatenation (.) or string at template
        test/scheduled_table.html.ep line 47 (#1)
Use of uninitialized value in join or string at
        lib/OpenQA/Schema/Result/Jobs.pm line 291 (#1)
Use of uninitialized value $build in concatenation (.) or string at
        lib/OpenQA/WebAPI/Plugin/Helpers.pm line 100 (#1)
Use of uninitialized value $distri in concatenation (.) or string at
        lib/OpenQA/WebAPI/Plugin/Helpers.pm line 100 (#1)
```

because job properties like `$job->MACHINE` are used uninitialized.

Also, it doesn't execute the `pitux@` test for me. It has just been scheduled for 15 minutes now and the test doesn't terminate. Just prints log messages like

```
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websocket 2
## removing timer setup_websocket
## adding timer setup_websinsgesamt 16
drwxr-xr-x 2 martchus users 4096  3. Jan 12:53 config
drwxr-xr-x 2 martchus users 4096  3. Jan 12:54 db
drwxr-xr-x 4 martchus users 4096  3. Jan 12:54 openqa
drwxr-xr-x 3 martchus users 4096  3. Jan 12:53 pool
insgesamt 16
drwxr-xr-x 2 martchus users 4096  3. Jan 12:53 config
drwxr-xr-x 2 martchus users 4096  3. Jan 12:54 db
drwxr-xr-x 4 martchus users 4096  3. Jan 12:54 openqa
drwxr-xr-x 3 martchus users 4096  3. Jan 12:53 pool
insgesamt 16
drwxr-xr-x 2 martchus users 4096  3. Jan 12:53 config
drwxr-xr-x 2 martchus users 4096  3. Jan 12:54 db
drwxr-xr-x 4 martchus users 4096  3. Jan 12:54 openqa
drwxr-xr-x 3 martchus users 4096  3. Jan 12:53 pool
insgesamt 16
drwxr-xr-x 2 martchus users 4096  3. Jan 12:53 config
drwxr-xr-x 2 martchus users 4096  3. Jan 12:54 db
drwxr-xr-x 4 martchus users 4096  3. Jan 12:54 openqa
drwxr-xr-x 3 martchus users 4096  3. Jan 12:53 pool
insgesamt 16
```

Do I need to call this test in a special way, maybe? Just tried `FULLSTACK=1 prove -v t/full-stack.t`.

File: t/full-stack.t (hunk 1 above)
Comment (coolo) [inline t/full-stack.t]: I had it used in the line below, but removed the usage. So no

**Comment by coolo** [comment]
I'm struggling with the same actually. The test only sleps 100 seconds and then aborts. So the little coverage gain we see is for the startup.

**Comment by coolo** [comment]
OK, works now. Unfortunately isotovideo runs under coverage too as we set PERL5OPTS and I don't want to unset it uncondtionally.

So I had select_re lib/.* to avoid os-autoinst/* and t/data/ perl code polluting the coverage data. This is not a regression though as before we only covered files in lib/. The -1 in files above is the current migration script - which we don't cover in general.

There is a lot more to cover/refactor now that we can :)

**Pull request merged**